File: 13-02-snackbar/java-13-02-snackbar/app/src/main/java/com/example/snackbar/MainActivity.java
package com.example.snackbar;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import androidx.appcompat.app.AppCompatActivity;
import androidx.coordinatorlayout.widget.CoordinatorLayout;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final CoordinatorLayout coordinatorView = findViewById(R.id.coordinatorLayout);
        final FloatingActionButton button = findViewById(R.id.button);
        button.setOnClickListener(view ->
                Snackbar.make(coordinatorView, "Hello Snackbar", Snackbar.LENGTH_SHORT)
                        .setAction("Another Snackbar", view1 ->
                                Snackbar.make(coordinatorView, "Hello Another Snackbar", Snackbar.LENGTH_SHORT)
                                        .show()
                        ).show());
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: 11-02-intent-service/kotlin-11-02-intent-service /app/src/main/java/com/example/intentservice/MainActivity.kt
package com.example.intentservice

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<View>(R.id.plus10).setOnClickListener { v ->
            val intent = Intent(this@MainActivity, FirstIntentService::class.java)
            intent.putExtra(FirstIntentService.NUMBER, 10)
            startService(intent)
        }

        findViewById<View>(R.id.minus10).setOnClickListener { v ->
            val intent = Intent(this@MainActivity, FirstIntentService::class.java)
            intent.putExtra(FirstIntentService.NUMBER, -10)
            startService(intent)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId


        return if (id == R.id.action_settings) {
            true
        } else super.onOptionsItemSelected(item)

    }
}
